spec/unit/pops/lookup/context_spec.rb

```diff
@@ -171,13 +171,8 @@ module Lookup
 
       context 'and multiple compilations' do
 
-        before(:each) { Puppet.settings[:environment_timeout] = 'unlimited' }
-        after(:each) do
-          Puppet.settings[:environment_timeout] = 0
-          Puppet.lookup(:environments).clear_all
-        end
-
         it 'will reuse cached_file_data and not call block again' do
+          Puppet.settings[:environment_timeout] = 'unlimited'
```

Comment:
Found the problem. The time resolution is not high enough to detect the change between the two compiles and neither the inode nor the size of the file changes. I'll push a PR that changes the content so that the size differs.